homeassistant/components/zwave_js/discovery_data_template.py

```diff
@@ -357,21 +358,11 @@ class NumericSensorDataTemplateData:
     unit_of_measurement: str | None = None
 
 
-T = TypeVar(
-    "T",
-    MultilevelSensorType,
-    MultilevelSensorScaleType,
-    MeterScaleType,
-    EnergyProductionParameter,
-    EnergyProductionScaleType,
-)
-
-
 class NumericSensorDataTemplate(BaseDiscoverySchemaDataTemplate):
     """Data template class for Z-Wave Sensor entities."""
 
     @staticmethod
-    def find_key_from_matching_set(
+    def find_key_from_matching_set[T: Enum](
         enum_value: T, set_map: Mapping[str, list[T]]
     ) -> str | None:
```

Comment:
The method works fine with any `Enum`. No need to constraint it to a list of specific ones.